crates/kubelet/src/kubelet.rs

```diff
@@ -295,16 +240,10 @@ mod test {
 
     #[async_trait::async_trait]
     impl Provider for MockProvider {
+        type InitialState = crate::state::Stub;
         const ARCH: &'static str = "mock";
-        async fn add(&self, _pod: Pod) -> anyhow::Result<()> {
-            Ok(())
-        }
-        async fn modify(&self, _pod: Pod) -> anyhow::Result<()> {
-            Ok(())
-        }
-        async fn delete(&self, _pod: Pod) -> anyhow::Result<()> {
-            Ok(())
-        }
+        async fn modify(&self, _pod: Pod) {}
```

Comment:
How do you add a new pod to the Provider if no `add` function is available?

Reply:
A new pod is created by running the state machine to completion. modify and delete remain useful.

Reply:
Yeah, this is something we talked about quite a bit. Basically each pod's machine is held in memory, running the pod until it has reached the end of its lifetime. Modify handles the cases where you get a deletion timestamp and the like